core/tests/test_utils_test.py

```diff
@@ -425,6 +425,47 @@ def mock_getcwd():
                 SwapWithCheckTestClass.getcwd_function_without_args()
 
 
+class EmailMockTests(test_utils.EmailTestBase):
+    """Class for testing EmailTestBase."""
+
+    def test_override_run_swaps_contexts(self):
+        """Assert that the feconf context swaps that wrap the default run
+        function correctly swaps the contexts.
+        """
+        self.assertEqual(feconf.MAILGUN_API_KEY, 'key')
+        self.assertEqual(feconf.MAILGUN_DOMAIN_NAME, 'name')
+
+    def test_mock_send_email_to_recipients_sends_correct_emails(self):
+        """Test sending email to recipients using mock adds the correct objects
+        to emails_dict.
+        """
+        self._send_email_to_recipients(
+            sender_email='a@a.com',
+            recipient_emails=['b@b.com'],
+            subject=(
+                'Hola 😂 - invitation to collaborate'
+                .encode(encoding='utf-8')),
+            plaintext_body='plaintext_body 😂'.encode(encoding='utf-8'),
+            html_body='Hi abc,<br> 😂'.encode(encoding='utf-8'),
+            bcc=['c@c.com'],
+            reply_to='abc',
+            recipient_variables={'b@b.com': {'first': 'Bob', 'id': 1}})
+        messages = self._get_sent_email_messages(
+            to='b@b.com')
+        all_messages = self._get_all_sent_email_messages()
+        self.assertEqual(len(messages), len(all_messages))
```

Comment:
Here and elsewhere, always prefer exact comparisons for the expected value. Somewhere, you should have a check that these are both equal to 1.

Reply:
Done